src/utils.zig

```diff
@@ -1919,6 +1919,20 @@ pub const BinaryReader = struct {
 		return std.mem.readInt(T, self.remaining[0..bufSize], self.endian);
 	}
 
+	pub fn readFloat(self: *BinaryReader, comptime T: type) error{OutOfBounds, IntOutOfBounds}!T {
+		const bufSize = @divExact(@typeInfo(T).float.bits, 8);
+		if(self.remaining.len < bufSize) return error.OutOfBounds;
+		defer self.remaining = self.remaining[bufSize..];
+
+		const intT = switch(T) {
+			f32 => u32,
+			f64 => u64,
+			else => unreachable,
```

Comment:
To make this more future proof, I'd prefer if you'd use `std.meta.Int`.

Reply:
That's great! I was looking for sth like this, I really need to save info about existance of `std.meta` somewhere C: